Client/mods/deathmatch/logic/luadefs/CLuaDrawingDefs.cpp

```diff
@@ -2121,3 +2122,22 @@ bool CLuaDrawingDefs::DxDrawWiredSphere(lua_State* const luaVM, const CVector po
     g_pCore->GetGraphics()->DrawWiredSphere(position, radius, color.value(), lineWidth.value_or(1), iterations.value_or(1));
     return true;
 }
+
+bool CLuaDrawingDefs::DxDrawModel3D(uint32_t modelID, CVector position, CVector rotation, const std::optional<CVector> scale, const std::optional<EModelLoadingScheme> loadingScheme)
```

Comment:
`std::uint32_t`

Reply:
`std` namespace isn't required by the official style guide. See https://github.com/multitheftauto/mtasa-blue/wiki/Style-Guide

Reply:
It was moved to [new docs](https://github.com/multitheftauto/mtasa-docs/blob/main/mtasa-blue/CODING_GUIDELINES.md)

Reply:
But this is still recommendation, right?

Reply:
The new guidelines are to be followed. Not really a recommendation anymore

Reply:
That's sad that this was adopted in docs. This is bad point and I'll insist on excluding it from guidlines. But now my goal is to merge this PR, not to argue about code style. So okay, I'll made the changes.

Reply:
The conclusions the was drawn from the discussions sound unconvincing. Nothing prevents us from mixing C++ and C, this is okay. `std::int32_t` is twice as long as `int32_t` and this is important.

Reply:
Its more apparent. And we already had typedefs in `SharedUtil.IntTypes.h` that makes it much more confusing.
Going with std types doesnt cause any confusion. 5 characters (`std::`) isnt much for that transparency.
WE already have much more bigger types. STD int types arent one of them

Reply:
SharedUtil.IntTypes.h is beautiful. It should be use wherever possible. See for example how it's implemented in UE: https://dev.epicgames.com/documentation/en-us/unreal-engine/epic-cplusplus-coding-standard-for-unreal-engine?application_version=5.3 (**Portable C++ code section**). Do you think they don't know what they're doing?

Reply:
STD types are crossplatform and guaranteed to have the size they describe

Reply:
SharedUtil.IntTypes.h is also can be crossplatform with just several fixes.

Reply:
Additional work needs to be done with no benefit.

Reply:
Benefit is the shorter types. The is important. Because code should be compact and readable. `std::int32_t` suffix neither compact not readable.

Reply:
It is compact though. just because it has `std::` prefix doesnt make this not readable. Would you find `std::string` unreadable? or `std::regex`?

Reply:
Yes, `std::string` is less readable than just `string`. This is why you can easily find aliases in an industrial(business, gamedev and so on) code. Even in MTA `SString` is much much more readable than `std::string`.

Reply:
I'd tend to argue, but ig its a matter of preference.
For me `std::string` is just as readable as string and even more due to its nature.

Reply:
We have our own guidelines now. No need to look at other projects anymore.

Reply:
Do you think our guidelines is perfect now?

Reply:
I think its decent enough.

Reply:
As I already have demonstrated they're not.

Reply:
because of `std::`? Thats not really a demonstration but again, a matter of preference.
Guidelines use std types and so should we.
Any further discussion about it should be redirected to discord instead of here.